### PR TITLE
Design test config and integrate in CC tests

### PR DESCRIPTION
**Pull request by vurusovs**

### CODE REVIEW

File: tests/conditional_compilation/test_config.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Using models from https://github.com/openvinotoolkit/testdata
+# $find models -wholename "*.xml"
+
+- model:
+    path: ${SHARE}/models/mobilenet_v2_1.4_224/mobilenet_v2_1.4_224_i8.xml
```
Comment (asomsiko) [inline tests/conditional_compilation/test_config.yml:5]: please replace `SHARE` with `TESTDATA` to make repository name clear

**Review by asomsiko — CHANGES_REQUESTED** [review]

File: tests/conditional_compilation/test_config.yml (hunk 1 above)
Comment (vurusovs) [reply tests/conditional_compilation/test_config.yml]: Done

File: tests/conditional_compilation/conftest.py [hunk 2]
```diff
@@ -17,32 +17,30 @@
...
-]
+from path_utils import expand_env_vars  # pylint: disable=import-error
 
 
 def pytest_addoption(parser):
     """ Define extra options for pytest options
     """
     parser.addoption(
-        "--models_root", required=True, type=Path, help="Path to models root directory"
+        "--test_conf",
```
Comment (vurusovs) [inline tests/conditional_compilation/conftest.py:34]: fix description in file header

**Review by asomsiko — CHANGES_REQUESTED** [review]

File: tests/conditional_compilation/conftest.py [hunk 3]
```diff
@@ -65,14 +63,17 @@ def pytest_generate_tests(metafunc):
...
         extra_args = {}
-        path = test["path"]
+        model_path = expand_env_vars(test["model"]["path"])
         if "marks" in test:
             extra_args["marks"] = test["marks"]
 
-        params.append(pytest.param(Path(path), **extra_args))
-        ids = ids + [path]
+        params.append(pytest.param(Path(model_path), **extra_args))
+        ids = ids + [model_path]
```
[the remark this answers is not in the document]
Comment (vurusovs) [reply tests/conditional_compilation/conftest.py:76]: Good catch! Fixed

File: tests/conditional_compilation/conftest.py (hunk 2 above)
Comment (vurusovs) [reply tests/conditional_compilation/conftest.py:34]: Done

**Review by asomsiko — APPROVED** [review]

File: tests/conditional_compilation/test_config.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Using models from https://github.com/openvinotoolkit/testdata
+# $find models -wholename "*.xml"
+
+- model:
```
Comment (vurusovs) [inline tests/conditional_compilation/test_config.yml:4]: Need to redesign test config to support several models in scope of one test case

Comment (asomsiko) [reply tests/conditional_compilation/test_config.yml:4]: Let's make it separate commit

**Pull request merged**